projects/engine/doc/api/manager.md

```diff
@@ -0,0 +1,291 @@
+
+_Only the caller plugn can update its profile._
+
+_The properties "name" and "url" cannot be updated._
+
+```typescript
+const methods = [ ...solidity.methods, 'serviceMethod' ]
+await solidity.call('manager', 'updateProfile', { methods })
+```
+
+### isActive
+Verify if a plugin is currently active.
+
+```typescript
+const isActive = await manager.isActive('solidity')
+```
+
+### activatePlugin
+Check if caller can activate plugin and activate it if authorized.
+
+_This method call `canActivate` under the hood._
+
+It can be called directly on the `PluginManager`:
+```typescript
+manager.activatePlugin('solidity')
+```
+
+Or from another plugin (for dependancy for example) :
+```typescript
+class EthDoc extends Plugin {
+  onActivation() {
+    return this.call('manager', 'activatePlugin', ['solidity', 'remix-tests'])
+  }
+}
+```
+
+### deactivatePlugin
+Check if caller can deactivate plugin and deactivate it if authorized.
+
+_This method call `canDeactivate` under the hood._
+
+It can be called directly on the `PluginManager`:
+```typescript
+manager.deactivatePlugin('solidity')
+```
+
+Or from another plugin :
+```typescript
+class EthDoc extends Plugin {
+  onDeactivation() {
+    return this.call('manager', 'deactivatePlugin', ['solidity', 'remix-tests'])
+  }
+}
+```
+
+> Deactivating a plugin can have side effect on other. We recommand to limit the access to this methods to a small set of plugins if any (see `canDeactivate`).
+
+### toggleActive
+Activate or deactivate by bypassing permission.
+
+_This method should ONLY be used by the IDE. Do not expose this method to other plugins._
+
+```typescript
+manager.toggleActive('solidity') // Toggle One
+manager.toggleActive(['solidity', 'remix-tests'])  // Toggle Many
+```
+
+## Permission
+
+By extending the `PluginManager` you can override the permission methods to create your own rules.
+
+### canActivate
+Check if a plugin can activate another.
+
+**Params**
+- `from`: The profile of the caller plugin.
+- `to`: The profile of the target plugin.
+
+```typescript
+class RemixManager extends PluginManager {
+  // Ask permission to user if it's not a plugin from Remix
+  async canActivate(from: Profile, to: Profile) {
+    if (this.isFromRemix(from.name)) {
+      return true
+    } else {
+      return confirm(`Can ${from.name} activates ${to.name}`)
+    }
+  }
+}
+```
+
+> Don't forget to let 'manager' activate plugins if you're not using `toggleActivate`.
+
+### canDeactivate
+Check if a plugin can deactivate another.
+
+**Params**
+- `from`: The profile of the caller plugin.
+- `to`: The profile of the target plugin.
+
+```typescript
+class RemixManager extends PluginManager {
+  // Only "manager" can deactivate plugins
+  async canDeactivate(from: Profile, to: Profile) {
+    return from.name === 'manager'
+  }
+}
+```
+
+
+> Don't forget to let 'manager' deactivate plugins if you're not using `toggleActivate`.
+
+
+### canCall
```

Comment:
is this `askUserPermission` ?

Reply:
In version 0.1.0 yes. Since version 0.2.0 we separate askUserPermission into tow method: 
- `canActivate`: if plugin can ativate another
- `canCall`: If a plugin can call a method from another.